lib/pbench/server/api/resources/datasets_inventory.py

```diff
@@ -76,16 +76,12 @@ def _get(
             # NOTE: we could choose to be "smarter" based on file size, file
             # type codes (e.g., .csv, .json), and so forth.
             return send_file(
-                file_stream,
+                file_info["stream"],
                 as_attachment=target is None,
                 download_name=file_info["name"],
             )
-        elif file_info["type"] != CacheType.FILE:
+        else:
             raise APIAbort(
                 HTTPStatus.UNSUPPORTED_MEDIA_TYPE,
                 "The specified path does not refer to a regular file",
             )
```

Comment:
Given we don't have complete agreement, I'm not sure how far I want to push that now; although a general cleanup of server HTTP errors would be great.

The only ones that actually make any possible sense here in terms of the RFC descriptions would be `FORBIDDEN` or `UNPROCESSABLE_ENTITY`; and I'm actually not that sure about either of those. (`FORBIDDEN`, for example, starts out by saying clearly that the request is fine but we're just refusing to process it: it does, however, go on to suggest that if the caller supplied authentication *then* it means that authorization was denied, which is really inconvenient as we generally expect that authentication would have been provided here...)

As much as I really hate it, and not least of all because it's so broadly defined and vastly overused, `BAD_REQUEST` with a good message string might be the best choice after all. The "due to something that is perceived to be a client error" language gives us fairly broad discretion here, but at the same token means we're not really saying much. Perhaps that's ultimately better than saying too much in a way that's easily misunderstood. 😦

Reply:
I can make peace with `BAD_REQUEST`....  👃 🤏